lib/puppet/provider/mysql_user/mysql.rb

```diff
@@ -123,11 +125,7 @@ def password_hash=(string)
     else
       # Version >= 5.7.6 (many password related changes)
       if (mysqld_type == "mysql" or mysqld_type == "percona") and Puppet::Util::Package.versioncmp(mysqld_version, '5.7.6') >= 0
-        if string.match(/^\*/)
-          mysql([defaults_file, system_database, '-e', "ALTER USER #{merged_name} IDENTIFIED WITH mysql_native_password AS '#{string}'"].compact)
-        else
-          raise ArgumentError, "Only mysql_native_password (*ABCD...XXX) hashes are supported"
```

Comment:
Just eyeballing it, but it looks like this removes validation for accidentally passing in non-hash passwords, which could cause confusion later down the road. Perhaps the validation should still be performed unless `@resource[:plugin] == 'sha256_password'`?